pkg/controllers/management/auth/auth_config.go

```diff
@@ -33,29 +34,46 @@ type CleanupService interface {
 }
 
 type authConfigController struct {
-	users            v3.UserLister
-	authRefresher    providerrefresh.UserAuthRefresher
-	cleanup          CleanupService
-	authConfigClient controllers.AuthConfigClient
+	users         v3.UserLister
+	authRefresher providerrefresh.UserAuthRefresher
+	cleanup       CleanupService
+	authConfigs   objectclient.GenericClient
 }
 
 func newAuthConfigController(context context.Context, mgmt *config.ManagementContext, scaledContext *config.ScaledContext) *authConfigController {
 	controller := &authConfigController{
-		users:            mgmt.Management.Users("").Controller().Lister(),
-		authRefresher:    providerrefresh.NewUserAuthRefresher(context, scaledContext),
-		cleanup:          cleanup.NewCleanupService(mgmt.Core.Secrets(""), mgmt.Wrangler.Mgmt),
-		authConfigClient: mgmt.Wrangler.Mgmt.AuthConfig(),
+		users:         mgmt.Management.Users("").Controller().Lister(),
+		authRefresher: providerrefresh.NewUserAuthRefresher(context, scaledContext),
+		cleanup:       cleanup.NewCleanupService(mgmt.Core.Secrets(""), mgmt.Wrangler.Mgmt),
+		authConfigs:   scaledContext.Management.AuthConfigs("").ObjectClient().UnstructuredClient(),
 	}
 	return controller
 }
 
 func (ac *authConfigController) setCleanupAnnotation(obj *v3.AuthConfig, value string) (runtime.Object, error) {
-	if obj.Annotations == nil {
-		obj.Annotations = make(map[string]string)
+	name := obj.Name
+	o, err := ac.authConfigs.Get(name, v1.GetOptions{})
+	if err != nil {
+		return nil, err
 	}
-	objCopy := obj.DeepCopy()
-	objCopy.Annotations[CleanupAnnotation] = value
-	return ac.authConfigClient.Update(objCopy)
+	u, ok := o.(runtime.Unstructured)
+	if !ok {
+		logrus.Errorf("auth config %s is not an unstructured value", name)
+		return nil, nil
```

Comment:
I don't think that returning `nil, nil` here is the right choice. At minimum, the sync method should be the one logging the error and choosing if the controller should re-enqueue.  This return is particularly dangerous because it returns a nil value for both the error and the object - one of those should be filled in. 

From a higher-level perspective here, I think returning nil for an error is deceptive for this function. We asked it to set the annotation, and it wasn't able to because of an unexpected condition. IMO that should result in an error.